lib/metrics/index.js

```diff
@@ -0,0 +1,147 @@
+'use strict';
+
+const PrometheusClient = require('./prometheus');
+const StatsDClient = require('./statsd');
+const LogClient = require('./log');
+const Metric = require('./metric');
+const normalizeName = require('./utils').normalizeName;
+
+const SUPPORTED_CLIENTS = {
+    PROMETHEUS: 'prometheus',
+    STATSD: 'statsd',
+    LOG: 'log'
+};
+
+const DEPRECATED_METHODS = ['timing', 'increment', 'decrement', 'gauge', 'unique'];
+
+class Metrics {
+    constructor(options, logger) {
+        this.logger = logger;
+        this.cache = new Map();
+        this.clients = [];
+        options.forEach((o) => {
+            switch (o.type) {
+                case SUPPORTED_CLIENTS.PROMETHEUS:
+                    this.clients.push(new PrometheusClient(o, this.logger));
+                    break;
+                case SUPPORTED_CLIENTS.STATSD:
+                    this.clients.push(new StatsDClient(o, this.logger));
+                    break;
+                case SUPPORTED_CLIENTS.LOG:
+                    this.clients.push(new LogClient(o, this.logger));
+                    break;
+                default:
```

Comment:
`log` is actually the default, let's keep it that way by having it if no spec has been given.

Reply:
Added in latest revision